CMakeLists.txt

```diff
@@ -397,8 +397,27 @@ endif(WITH_RADOSGW)
 
 
 if (WITH_RADOSGW)
+# Maybe we can use GetPrerequisites to do this if we have a matrix of various
+# ssl libs over distros, https://curl.haxx.se/docs/install.html mentions the
+# configure flags for non openssl libs
+  execute_process(
+    COMMAND
+  "sh" "-c"
+  "curl-config --configure | grep with-ssl"
+  RESULT_VARIABLE NO_CURL_SSL_LINK
+  ERROR_VARIABLE CURL_CONFIG_ERRORS
+  )
+  if (CURL_CONFIG_ERRORS)
+    message(WARNING "unable to run curl-config; rgw cannot make ssl requests to external systems reliably")
+  endif()
   find_package(OpenSSL)
   if (OPENSSL_FOUND)
+    if (NOT NO_CURL_SSL_LINK)
+      if (OPENSSL_VERSION VERSION_LESS "1.1.0")
+        message(STATUS "Found openssl ${OPENSSL_VERSION} < 1.1.0 need to explicitly set locking primitives")
+        set(WITH_RADOSGW_OPENSSL ON)
```

Comment:
this define is a lot more specific than 'radosgw uses openssl' - maybe something like `WITH_LIBCURL_OPENSSL_LOCK_COMPAT`

Reply:
ack much better